src/OpenTelemetry/Trace/TraceIdRatioBasedSampler.cs

```diff
@@ -66,6 +66,8 @@ public TraceIdRatioBasedSampler(double probability)
         /// <inheritdoc />
         public override SamplingResult ShouldSample(in SamplingParameters samplingParameters)
         {
+            if (idUpperBound == long.MinValue) return new SamplingResult(false);
```

Comment:
This would not help in most cases as the expected usage of `TraceIdRatioBasedSampler` when the probability is set to something in (0.0,1.0) (excluding 0.0 and 1.0: we have dedicated `AlwaysOffSampler` and `AlwaysOnSampler` for them respectively). In fact, this is only going to lower the performance by adding two more if checks in the hot-path for most of the cases.

Reply:
@WeihanLi how important is this optimization?
If it is indeed a common path with good perf yield, this could be the right place to put the logic https://github.com/open-telemetry/opentelemetry-dotnet/blob/1143c52e739a8af22da0ccd213ba6ba16b8f9241/src/OpenTelemetry/Trace/TracerProviderSdk.cs#L234.

Reply:
We're using the `TraceIdRatioBasedSampler` in our code, and for the test environment, we're using `1.0` for the sampling rate, while for production maybe `0.1` or `0.01` would be used, for this case I think it's a point can be improved